keyboards/junco/junco_sync.c

```diff
@@ -0,0 +1,286 @@
+// Copyright 2022 Dane Skalski (@Daneski13)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+/* Only relevant if junco sync is enabled */
+#ifdef JUNCO_SYNC_ENABLE
+#    include "junco_sync.h"
+#    include "print.h"
+#    include "transactions.h"
+#    include "split_util.h"
+#    include "junco.h"
+
+/* Sync common keyboard data */
+
+// Pointer to the user's user_config.raw. If the user wants to sync their own user config,
+// set this to the address of user_config.raw in the keyboard_post_init after reading the
+// user_config from eeprom `user_config_raw_ptr = &user_config.raw`.
+uint32_t* user_config_raw_ptr = NULL;
+
+// Data type for sending data to slave
+typedef struct {
+    uint32_t kb_conf;     // Current keyboard config
+    uint32_t default_lay; // Current default layer, SPLIT_LAYER_STATE_ENABLE is not persistent and may not sync in time
+    uint32_t user_conf;   // If the user has a config...
+    uint16_t keymap_conf; // Current keymap config
+} kb_sync_data_t;
+// Saves common data on master and sends it to slave
+__attribute__((weak)) bool junco_sync(void) {
+    // Count how many times retried
+    static uint8_t retries = 0;
+    // Allow max retries
+    if (retries >= JUNCO_SYNC_RETRIES) {
+        dprintf("Failure to sync: too many retries\n");
+        retries = 0;
+        return true;
+    }
+
+    // Ensure a split is actually connected
+    if (!is_transport_connected()) {
+        dprintf("Failure to sync: No split detected\n");
+        retries++;
+        return false;
+    }
+
+    uint32_t layer = get_highest_layer(default_layer_state);
```

Comment:
uint8_t is all you need for this.  layer state is 32 bit at most, so the highest layer will only ever be 32.  So ... uint8_t is more than enough.